samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -204,7 +212,15 @@ public synchronized void send(String source, OutgoingMessageEnvelope envelope) {
     checkCallbackThrowable("Received exception on message send");
 
     EventData eventData = createEventData(streamId, envelope);
-    int eventDataLength =  eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    int eventDataLength = eventData.getBytes() == null ? 0 : eventData.getBytes().length;
+    if (eventDataLength > this.maxMessageSize) {
+      LOG.info("Received a message with size {} > maxMessageSize configured {(}), Skipping it", eventDataLength,
```

Comment:
I have a little bit concern about this. We never know whether one day we will be streaming a pipeline that has many many large messages. Some naive restriction on the per event logging would be nice.

Reply:
Logging the skipped message is a requirement from the customer, So keeping it as is.